test/test_torch.py

```diff
@@ -10069,6 +10069,15 @@ def test_unfold_all_devices_and_dtypes(self, device):
                 x = torch.randint(5, (0, 1, 3, 0), dtype=dt, device=device)
                 self.assertEqual((0, 1, 1, 0, 3), x.unfold(2, 3, 2).shape)
 
+    def test_unfold_scalars(self, device):
+        x = torch.tensor(0.5, device=device)
+        # unfold on a 0-dimensional tensor should always return a 1-d dimensional
+        # tensor of shape [size] (i.e., the second parameter to unfold)
```

Comment:
Does it actually work if you do `x.unfold(0, 2, 1)` ?
The only valid values for size are 0 and 1 here right?

Reply:
Now, it will give you a error.